cubedash/_stac.py

```diff
@@ -0,0 +1,235 @@
+import logging
+
+from collections import OrderedDict
+from flask import Blueprint, request, abort, url_for
+
+from datacube.api.query import Query
+from datacube.model import Range
+from datacube.utils.geometry import CRS, Geometry
+
+import datetime
+from dateutil.parser import parse
+from pandas import Timestamp, DateOffset
+import yaml
+import json
+import itertools
+
+from . import _utils as utils
+from . import _model
+
+
+_LOG = logging.getLogger(__name__)
+bp = Blueprint('stac', __name__, url_prefix='/stac')
+
+CFG = yaml.load(
+    """
+    homepage: "http://www.ga.gov.au/"
+
+    license:
+      name: "Creative Commons Attribution 4.0 International"
+      link: "https://spdx.org/licenses/CC-BY-4.0.html#licenseText"
+      short_name: "CC-BY-4.0"
+      copyright: "DEA, Geoscience Australia"
+
+    contact:
+      name: "Geoscience Australia"
+      organization: "Commonwealth of Australia"
+      email: "sales@ga.gov.au"
+      phone: "+61 2 6249 9966"
+      url: "http://www.ga.gov.au"
+    """
+)
+
+MAX_DATASETS = 100
+DATASETS_PER_REQUEST = 20
+
+
+@bp.route('/')
+def root():
+    return utils.as_json({})
+
+
+@bp.route('/search', methods=['GET', 'POST'])
+def stac_search():
+    if request.method == 'GET':
+        bbox = request.args.get('bbox')
+        time_ = request.args.get('time')
+        product = request.args.get('product')
+        limit = request.args.get('limit')
+        from_dts = request.args.get('from')
+    else:
+        req_data = request.get_json()
+        bbox = req_data.get('bbox')
+        time_ = req_data.get('time')
+        product = req_data.get('product')
+        limit = req_data.get('limit')
+        from_dts = req_data.get('from')
+
+    # bbox and time are compulsory
+    if not bbox:
+        abort(404, "bbox must be specified")
+    if not time_:
+        abort(404, "time must be specified")
+
+    # Verify and cast data types of request data
+    bbox = json.loads(bbox) if isinstance(bbox, str) else bbox
+    time_ = time_ if isinstance(time_, str) else json.dumps(time_)
+    limit = int(limit) if isinstance(limit, str) else limit
+    if from_dts and isinstance(from_dts, str):
+        from_dts = int(from_dts)
+
+    # from_dts must be lower than limit
+    limit = MAX_DATASETS if not limit else min(limit, MAX_DATASETS)
+
+    if from_dts and from_dts >= limit:
```

Comment:
I wanted to keep `from_dts` within `limit` specified so that `paging` occur only within `limit`

Reply:
Wanted to keep paging within `limit` specified since anyway we are truncating at `MAX_DATASETS`

Reply:
I think this is breaking from the stac spec in a way that will confuse users, and I can't see any benefits to doing it this way (it's not any simpler, right?).

If I understand it correctly:
- The stac spec behaviour is that specifying `limit=3`, I'll get 3 results on a page, and I can keep loading pages until we hit our own internal `MAX_DATASETS` limit (which we may remove in the future).
- With this behaviour, if I query with `limit=3`, I'll get 3 results and a later page will throw an error because it's higher than 3. Stac clients that already specify their own page sizes may not realise they're  limiting the total datasets they can see, and they will need to change it to 10000 to see all 10000 datasets up to the `MAX_DATASETS` limit (we don't want them to start doing this!).

(If I really want to limit myself to 3 results total, I can already do that by setting `limit=3` and  ignoring the next page; I don't need the second behaviour to do it for me.)

Reply:
Hmm OK I will change this. I had interpreted `limit` as total required rather than a possible page size